resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -81,7 +81,7 @@ private[yarn] class YarnAllocator(
   private val releasedContainers = Collections.newSetFromMap[ContainerId](
     new ConcurrentHashMap[ContainerId, java.lang.Boolean])
 
-  private val numExecutorsRunning = new AtomicInteger(0)
+  private val runningExecutors = new java.util.concurrent.ConcurrentHashMap[String, Unit]()
```

Comment:
This can be changed to `Collections.newSetFromMap`, since we only need `Set` instead of `Map`.